types/types.go

```diff
@@ -12,6 +18,71 @@ type EntityID = []byte
 
 type CensusRoot = []byte
 
+// VoterID is the indentifier of a voter, it is represented as random bytes
+// The first byte of the slice indicates one of the supported identifiers
+// For example for an Ethereum public key the VoterID is [1, pubkb0, pubkb1, ...]
+// where pubkb0 is the first byte of the Ethereum public key
+type VoterID []byte
+
+// VoterIDType represents the type of a voterID
+type VoterIDType = uint8
+
+const (
+	VoterID_Type_Undefined VoterIDType = 0
+	VoterID_Type_ECDSA     VoterIDType = 1
+)
+
+// Enum value maps for VoterIDType.
+var (
+	VoterID_Type_name = map[VoterIDType]string{
+		VoterID_Type_Undefined: "UNDEFINED",
+		VoterID_Type_ECDSA:     "ECDSA",
+	}
+	VoterID_Type_value = map[string]VoterIDType{
+		"UNDEFINED": VoterID_Type_Undefined,
+		"ECDSA":     VoterID_Type_ECDSA,
+	}
+)
+
+var (
+	ErrUnsupportedVoterIDType error = errors.New("voterID type not supported")
+)
+
+func (v VoterID) Type() VoterIDType {
+	if _, ok := VoterID_Type_name[v[0]]; !ok {
+		return VoterID_Type_Undefined
```

Comment:
Actually if you query a map key that does not exist, the value equals to the zero value. So you might not need this check and just return `VoterIDType(v[0])`